nilearn/decomposition/tests/test_canica.py

```diff
@@ -135,3 +135,12 @@ def test_component_sign():
                 canica.components_)):
             mp = mp.get_data()
             assert_less_equal(-mp.min(), mp.max())
+
+
+def test_threshold_bound():
+    data, mask_img, components, rng = _make_canica_test_data()
```

Comment:
We don't need this because directly validating the parameters where data is not required.